open-sphere-base/core/src/main/java/io/opensphere/core/util/fx/AutoSelectComboBox.java

```diff
@@ -0,0 +1,155 @@
+package io.opensphere.core.util.fx;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+
+import javafx.collections.ObservableList;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.ListView;
+import javafx.scene.control.TextField;
+import javafx.scene.control.skin.ComboBoxListViewSkin;
+import javafx.scene.input.KeyEvent;
+
+/**
+ * An extension to the combo box that allows the user to type and selects the
+ * first matching item. If the user then presses enter, the item is selected,
+ * the full text of the item is placed into the editor, and a selected event is
+ * fired.
+ *
+ * @param <T> the type of data stored in the combo box.
+ */
+public class AutoSelectComboBox<T> extends ComboBox<T>
+{
+    /**
+     * Creates a new combobox. All auto-selectable combo boxes are editable by
+     * default.
+     */
+    public AutoSelectComboBox()
+    {
+        initialize();
+    }
+
+    /**
+     * Creates a new combobox bound to the supplied observable list. All
+     * auto-selectable combo boxes are editable by default
+     * 
+     * @param items the items to show in the list.
+     */
+    public AutoSelectComboBox(ObservableList<T> items)
+    {
+        super(items);
+        initialize();
+    }
+
+    /** Initializes the combobox to be editable and registers all listeners. */
+    private void initialize()
+    {
+        setEditable(true);
+        setOnKeyTyped(evt -> handleKeyRelease(evt));
+        setOnKeyReleased(evt -> handleKeyRelease(evt));
+    }
+
+    /**
+     * Handles the release of the key described in the event.
+     * 
+     * @param event the event describing the key release.
+     */
+    public void handleKeyRelease(KeyEvent event)
```

Comment:
Duplicated into/from AutoSelectComboBoxListener?

Reply:
yep, I duplicated the code using a listener and a full up extension of the combo box to allow for use in either case.